### DISCUSSION TITLE
custom key generation for table component

### DISCUSSION THREAD

**Discussion by heyitzkamal** [mantinedev/mantine#6090] [Feature requests]
currently, the table component generates keys for table rows based on their index in the data array provided through the data prop but this approach leads to react warnings about missing unique keys when rendering lists. to address this issue and provide more flexibility, i propose the addition of a keyGenerator function to the table component's data prop. this function would allow users to specify a custom key for each row in the table.

**Comment by rtivital** [comment]
How can keys be duplicated if index is unique to each element in the array?
